core/system/src/mpu/vmpu.c

```diff
@@ -59,16 +59,24 @@ static int vmpu_sanity_checks(void)
     assert(vmpu_bits(0x8001UL) == 16);
     assert(vmpu_bits(1) == 1);
 
+    /* Verify that the known hard-coded symbols are equal to the ones taken from
+     * the host linker script. */
+    assert((uint32_t) __uvisor_config.flash_start == FLASH_ORIGIN);
+    assert((uint32_t) __uvisor_config.flash_offset == FLASH_ORIGIN + RESERVED_FLASH);
+    assert((uint32_t) __uvisor_config.sram_start == SRAM_ORIGIN);
+    assert((uint32_t) __uvisor_config.sram_offset == SRAM_ORIGIN + RESERVED_SRAM);
+
+    /* Verify that the uVisor binary blob is positioned at the flash offset. */
+    assert(__uvisor_config.flash_offset == __uvisor_config.main_start);
+
     /* verify that __uvisor_config is within valid flash */
-    assert( ((uint32_t) &__uvisor_config) >= FLASH_ORIGIN );
-    assert( ((((uint32_t) &__uvisor_config) + sizeof(__uvisor_config))
-             <= (FLASH_ORIGIN + FLASH_LENGTH)) );
+    assert(VMPU_FLASH_ADDR((uint32_t) &__uvisor_config));
+    assert(VMPU_FLASH_ADDR(((uint32_t) &__uvisor_config) + sizeof(__uvisor_config)));
 
     /* verify if configuration mode is inside flash memory */
-    assert((uint32_t)__uvisor_config.mode >= FLASH_ORIGIN);
-    assert((uint32_t)__uvisor_config.mode <= (FLASH_ORIGIN + FLASH_LENGTH - 4));
-    DPRINTF("uvisor_mode: %u\n", *__uvisor_config.mode);
-    assert(*__uvisor_config.mode <= 2);
+    assert(VMPU_FLASH_ADDR((uint32_t) __uvisor_config.mode));
+    assert(*(__uvisor_config.mode) <= 2);
+    DPRINTF("uVisor mode: %u\n", *(__uvisor_config.mode));
```

Comment:
Doesn't this mode get printed somewhere else already? Do we want to print it twice?